course_discovery/apps/api/v1/views/program_creation.py

```diff
@@ -71,14 +73,41 @@ class ProgramUploadView(APIView):
     parser_class = (ImageUploadParser,)
 
     def post(self, request, uuid):
-        if 'file_upload' not in request.data:
-            Response(status=status.HTTP_204_NO_CONTENT)
-        file = request.data['file_upload']
         try:
-            img = Image.open(file)
-            img.verify()
-            program = Program.objects.get(uuid=uuid)
+            file = request.data.get('file_upload')
+            program = Program.objects.filter(uuid=uuid).first()
+            if program and file:
+                img = Image.open(file)
+                img.verify()
+                program.banner_image.save(file.name, file, save=True)
+                return Response(status=status.HTTP_200_OK)
+            else:
+                return Response(status=status.HTTP_204_NO_CONTENT)
+        except:
+            return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class ProgramCardImageUploadView(APIView):
+    permission_classes = (DjangoModelPermissions,)
+    queryset = Program.objects.all()
+    parser_class = (ImageUploadParser,)
+
+    def post(self, request, uuid):
+        try:
+            file = request.data.get('file_upload')
+            program = Program.objects.filter(uuid=uuid).first()
+            if program and file:
+                CARD_IMAGE_URL = 'media/programs/card_images'
+                DIRECTORY = settings.MEDIA_ROOT + '/' + CARD_IMAGE_URL
+                if not os.path.exists(DIRECTORY):
+                    os.makedirs(DIRECTORY)
+                IMAGE = DIRECTORY + '/{}.png'.format(program.uuid)
```

Comment:
Hi @huynguyen-groove ,
Can you please rename variables name to lower cases?

- CARD_IMAGE_URL => program_card_path
- DIRECTORY => media_root_path
- IMAGE => image_path

Reply:
we need to keep option 1 for intelligibility, Thanks !